### DISCUSSION TITLE
GUI for pandapower

### DISCUSSION THREAD

**Discussion by manuvarkey** [e2nIEE/pandapower#1758] [Show and tell]
Hi all,

I have written a GUI for pandapower ([https://github.com/manuvarkey/GElectrical](https://github.com/manuvarkey/GElectrical)) if any one is interested. The application is intended for MV/LV distribution networks but may be useful for other applications also. Currently application is in beta release phase.

Any ideas for improvement are most welcome. Thank you.

**Comment by julffers** [comment]
Your GUI looks very interesting and I would like to try it out. Could you provide me with some details on how to install and start it?

**Comment by AnkurArohi** [comment]
I would like to use it, can you share some details about the installation , please

**Comment by manuvarkey** [comment]
Hi @julffers @AnkurArohi, installer files for windows and Linux are available in the following link.

[https://github.com/manuvarkey/GElectrical/releases/tag/v1-RC.1](https://github.com/manuvarkey/GElectrical/releases/tag/v1-RC.1)

**Comment by manuvarkey** [comment]
Updated version with support for power flow with diversity factor, bug fixes.
[https://github.com/manuvarkey/GElectrical/releases/tag/v1-RC.2](https://github.com/manuvarkey/GElectrical/releases/tag/v1-RC.2)

Linux version is now published at Flathub. 
[https://flathub.org/apps/details/com.kavilgroup.gelectrical](https://flathub.org/apps/details/com.kavilgroup.gelectrical)

**Reply by nicorikken** [reply]
I just tested it from Flatpak. It was really easy to get started. It takes a bit of time to understand the UI, how to run simulations and where to view the results.
Is there a way to look under the hood to the Python code for the simulation. As this is based on PandaPower, I could see the use-case of clicking a simulation in the UI and exporting the skeleton code to run the simulation. I know that would totally be a different use-case, but perhaps that would help those wanting to start with PandaPower to get going more easily.
 
How can you be helped to continue this development and ensure a bright feature of this software?

**Reply by manuvarkey** [reply]
Hi @nicorikken ,

I have created a small screen cast  showing the usage of the software.

[GElectrical v1-RC.3 (dev) screen cast ](https://youtu.be/Kq7LwS2ovFM)

Some user interface tweaks are required to make the software more user friendly. Adding more tool tips describing details of various options are in the pipeline. A proper documentation is required to be written for end users.

Future of the software depends on its usage and feedback by electrical & power engineers regarding what features needs implementation and what needs change. Also a better standard library of elements are needed (lines, breakers, fuses, cables, transformers etc).

**Reply by manuvarkey** [reply]
> Is there a way to look under the hood to the Python code for the simulation. As this is based on PandaPower, I could see the use-case of clicking a simulation in the UI and exporting the skeleton code to run the simulation. I know that would totally be a different use-case, but perhaps that would help those wanting to start with PandaPower to get going more easily.

The pandapower network used for the simulation is expoted as `network.json` file along with the analysis results. This can be loaded as follows.
```
import pandapower as pp
net = pp.from_json('network.json')
```

**Comment by manuvarkey** [comment]
Updated version
https://github.com/manuvarkey/GElectrical/releases/tag/v1-RC.3

**Comment by manuvarkey** [comment]
Now documentation for the application is hosted at https://gelectrical.readthedocs.io/
